File: src/main/java/net/andrecarbajal/naviMusic/commands/music/RepeatCommand.java
package net.andrecarbajal.naviMusic.commands.music;

import net.andrecarbajal.naviMusic.audio.GuildMusicManager;
import net.andrecarbajal.naviMusic.audio.MusicService;
import net.andrecarbajal.naviMusic.commands.SlashCommand;
import net.andrecarbajal.naviMusic.dto.response.Response;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
import org.springframework.stereotype.Component;

@Component
public class RepeatCommand extends SlashCommand {
    private final MusicService musicService;

    public RepeatCommand(MusicService musicService) {
        super("repeat", "Repeats the current track");
        this.musicService = musicService;
    }

    @Override
    public void onCommand(SlashCommandInteractionEvent event) {
        if (noVoiceChannelCheck(event)) return;
        event.deferReply().queue();

        GuildMusicManager guildMusicManager = musicService.getGuildMusicManager(event.getChannel().asTextChannel().getGuild());

        if (guildMusicManager.scheduler.isRepeating())
            new Response("Already repeating", Response.Type.ERROR, false).editReply(event);
        else {
            guildMusicManager.scheduler.setRepeating(true);
            new Response("Repeating the current track", Response.Type.OK, false).editReply(event);
        }
    }
}

File: src/main/kotlin/net/andrecarbajal/naviMusic/commands/music/RepeatCommand.kt
package net.andrecarbajal.naviMusic.commands.music

import net.andrecarbajal.naviMusic.audio.MusicService
import net.andrecarbajal.naviMusic.commands.SlashCommand
import net.andrecarbajal.naviMusic.dto.response.RichResponse
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import org.springframework.stereotype.Component

@Component
class RepeatCommand(private val musicService: MusicService) :
    SlashCommand("repeat", "Repeats the current track", Category.MUSIC) {

    override fun onCommand(event: SlashCommandInteractionEvent) {
        if (noVoiceChannelCheck(event)) return
        event.deferReply().queue()

        val guild = event.guild ?: return
        val guildMusicManager = musicService.getGuildMusicManager(guild)

        if (guildMusicManager.scheduler.isRepeating) {
            RichResponse("Already repeating", RichResponse.Type.ERROR, false).editReply(event)
        } else {
            guildMusicManager.scheduler.isRepeating = true
            RichResponse("Repeating the current track", RichResponse.Type.OK, false).editReply(event)
        }
    }
}
